generic/src/main/scala/io/finch/generic/package.scala

```diff
@@ -0,0 +1,8 @@
+package io.finch
+
+package object generic {
+  /**
+   * Generically derive a very basic instance of [[Endpoint]] for a given type `A`.
+   */
+  def deriveFor[A]: GenericDerivation[A] = new GenericDerivation[A]
```

Comment:
Since we're moving this anyway, how about we change that to `deriveEndpoint[A]`?